paasta_tools/deployd/master.py

```diff
@@ -286,9 +288,21 @@ def start_watchers(self):
         for watcher in self.watcher_threads:
             watcher.start()
         self.log.info("Waiting for all watchers to start")
-        while not all([watcher.is_ready for watcher in self.watcher_threads]):
-            self.log.debug("Sleeping and waiting for watchers to all start")
-            time.sleep(1)
+        attempts = 0
+        while attempts < 24:
+            if all([watcher.is_ready for watcher in self.watcher_threads]):
+                return
+            self.log.info("Sleeping and waiting for watchers to all start")
+            self.log.info([
+                {
+                    'name': watcher.__class__.__name__,
+                    'ready': watcher.is_ready,
+                } for watcher in self.watcher_threads
+            ])
+            time.sleep(5)
```

Comment:
this seems a long time? I'm kind of expecting these workers to start quickly - what could lead to them taking so long?

Reply:
Yeah, I wasn't quite sure what was reasonable. I think we have to get a watch on all the ZK nodes but I guess that is quick. Any reason not to be conservative though?